From e5b19f14eac609e3c315e14e20ca1935de8de0e3 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Wed, 13 Oct 2010 00:42:55 +0000
Subject: Revert "dix: don't create core motion events for non-x/y valuators."

Reverting due to fallout discussed in bug #30267.  See:
https://bugs.freedesktop.org/show_bug.cgi?id=30267

This reverts commit d4ef63f602325a9920dc1cbf64e3969dfa394d5f.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
diff --git a/dix/eventconvert.c b/dix/eventconvert.c
index 0f747c1..4e3de0b 100644
--- a/dix/eventconvert.c
+++ b/dix/eventconvert.c
@@ -102,15 +102,6 @@ EventToCore(InternalEvent *event, xEvent *core)
     switch(event->any.type)
     {
         case ET_Motion:
-            {
-                DeviceEvent *e = &event->device_event;
-                /* Don't create core motion event if neither x nor y are
-                 * present */
-                if (!BitIsOn(e->valuators.mask, 0) &&
-                    !BitIsOn(e->valuators.mask, 1))
-                    return BadMatch;
-            }
-            /* fallthrough */
         case ET_ButtonPress:
         case ET_ButtonRelease:
         case ET_KeyPress:
--
cgit v0.8.3-6-g21f6
